cli/installer/install-azd.ps1

```diff
@@ -118,39 +190,49 @@ Remove-Item $tempFolder -Recurse -Force | Out-Null
 # from the registry with the DoNotExpandEnvironmentNames option and write that
 # value back using the non-destructive [Environment]::SetEnvironmentVariable
 # which also broadcasts environment variable changes to Windows.
-if (!$NoPath) {
-    try {
-        $registryKey = [Microsoft.Win32.Registry]::CurrentUser.OpenSubKey('Environment', $false)
-        $originalPath = $registryKey.GetValue(`
-            'PATH', `
-            '', `
-            [Microsoft.Win32.RegistryValueOptions]::DoNotExpandEnvironmentNames `
-        )
-        $pathParts = $originalPath -split ';'
-
-        if (!($pathParts -contains $InstallFolder)) {
-            Write-Host "Adding $InstallFolder to PATH"
-
-            # SetEnvironmentVariable broadcasts the "Environment" change to
-            # Windows and is NOT destructive (e.g. expanding variables)
-            [Environment]::SetEnvironmentVariable(
+if (!$NoPath -and !(isLinuxOrMac)) {
+    # Wrap the Microsoft.Win32.Registry calls in a script block to prevent the
+    # type intializer from attempting to initialize those objects in non-Windows
+    # environments.
+    . {
+        try {
+            $registryKey = [Microsoft.Win32.Registry]::CurrentUser.OpenSubKey('Environment', $false)
```

Comment:
Try wrapping just the offended calls in script blocks. Curious if that also mitigates the issue.

Reply:
Wrapping just the `Microsoft.Win32` namespace calls works.